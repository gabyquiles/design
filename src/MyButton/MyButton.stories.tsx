import type { Meta, StoryObj } from '@storybook/react';
import { MyButton } from './MyButton';

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
const meta = {
    title: 'Atoms/Button',
    component: MyButton,
    parameters: {
      // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/react/configure/story-layout
      layout: 'centered',
    },
    // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/react/writing-docs/autodocs
    tags: ['autodocs'],
    // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
    argTypes: {
      onClick: { action: 'clicked'}
    },
  } satisfies Meta<typeof MyButton>;
  
  export default meta;

export const Primary: StoryObj = {
  args: {
    primary: true,
    children: 'Button',
  },
};