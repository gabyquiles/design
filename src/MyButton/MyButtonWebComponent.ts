import {html, css, LitElement} from 'lit';
import {customElement} from 'lit/decorators.js';

@customElement('my-button')
export class MyButton extends LitElement {
  static styles = css`
    button {
      padding: 5px 15px;
      font-size: 0.85rem;
      border: solid 1px rgb(0, 144, 255, 0.5);
      border-radius: 4px;
      background: transparent;
      color: rgb(0, 144, 255);
    }
    button:hover {
      background-color: rgb(0, 144, 255, 0.05);
      border: solid 1px rgb(0, 144, 255, 5);
    }
  `;

  render() {
      return html`
        <button @click="${this.handleClick}">
          <slot></slot>
        </button>
      `;
  }

  private handleClick(e: MouseEvent) {
    e.stopPropagation();
    this.dispatchEvent(new Event('click', {bubbles: true, composed: true}));
  }
}