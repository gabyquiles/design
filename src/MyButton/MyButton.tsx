import * as React from 'react';
import {createComponent} from '@lit/react';
import { MyButton as MyButtonWC } from './MyButtonWebComponent';

export const MyButton = createComponent<MyButtonWC>({
    react: React,
    tagName: 'my-button',
    elementClass: MyButtonWC,
    displayName: 'MyButton',
    events: {
      onClick: 'click'
    }
})

export interface MyButtonProps extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLElement>, HTMLElement> {
    value: string;
    /**
     * Optional click handler
     */
    onClick?: () => void;
}

declare global {
    namespace JSX {
      interface IntrinsicElements {
        'my-button': MyButtonProps
      }
    }
}

export default MyButton;