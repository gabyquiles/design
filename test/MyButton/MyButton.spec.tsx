import * as React from 'react';
import { render, waitFor } from '@testing-library/react';
import { screen } from 'shadow-dom-testing-library';

import { MyButton } from "../../src/MyButton"

describe('MyButton', () => {
    it('renders correctly', () => {
        const { container } = render(<MyButton>Click Me</MyButton>)
        expect(container).toMatchSnapshot();
    })
    it('shows the text for the button', async () => {
        render(<MyButton>Click Me</MyButton>)
        expect(screen.getByShadowText('Click Me')).toBeInTheDocument()
    })
})