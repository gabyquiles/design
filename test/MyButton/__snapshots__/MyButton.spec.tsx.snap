// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders correctly 1`] = `
<div>
  <my-button>
    Click Me
  </my-button>
</div>
`;

exports[`MyButton > renders correctly 1`] = `
<div>
  <my-button>
    Click Me
  </my-button>
</div>
`;
